apps/wallet/src/ui/app/redux/slices/transaction-requests/index.ts

```diff
@@ -65,10 +66,105 @@ export const loadTransactionResponseMetadata = createAsyncThunk<
     }
 );
 
+export type TxnMetaResponse = {
+    coinSymbol?: string | null;
+    amount?: number | null;
+    objectId?: string;
+    isListing?: boolean;
+} | null;
+
+const getRequestCost = (
+    txEffects: TransactionEffects,
+    address: string
+): TxnMetaResponse | null => {
+    const events = txEffects?.events || [];
+
+    let sumCoin = 0;
+    const coinMeta: {
+        objectId?: string;
+        amount?: number;
+        coinSymbol?: string;
+        isListing?: boolean;
+    } = {};
+
+    for (const event of events) {
+        if (
+            'coinBalanceChange' in event ||
+            'transferObject' in event ||
+            'moveEvent' in event
+        ) {
+            // Get the amount in move events
+            if (
+                event.moveEvent?.sender === address &&
+                event.moveEvent?.fields?.price
+            ) {
+                sumCoin += event.moveEvent.fields.price;
+                coinMeta.amount = sumCoin;
+                coinMeta.coinSymbol = '0x2::sui::SUI';
+                coinMeta.isListing = true;
+            }
+
+            // aggregate coin balance for pay
+            if (
+                event.coinBalanceChange?.changeType === 'Pay' &&
+                event.coinBalanceChange?.owner.AddressOwner === address
+            ) {
+                sumCoin += event.coinBalanceChange?.amount || 0;
+                coinMeta.amount = sumCoin;
+                coinMeta.coinSymbol = event.coinBalanceChange?.coinType
+                    ? event.coinBalanceChange.coinType
+                    : '';
+            }
+
+            // TODO - support multiple objectIds
+            if (event.transferObject?.recipient?.AddressOwner === address) {
+                coinMeta.objectId = event.transferObject.objectId;
+            }
+        }
+    }
+    return coinMeta || null;
+};
+
+export const executeDryRunTransactionRequest = createAsyncThunk<
+    {
+        txRequestID: string;
+        txnMeta?: TxnMetaResponse;
+        txGasEstimation?: number;
+    },
+    {
+        txData: string | SignableTransaction | Base64DataBuffer;
+        id: string;
+    },
+    AppThunkConfig
+>(
+    'dryrun-transaction-request',
+    async (data, { getState, extra: { api, keypairVault } }) => {
+        const { txData, id } = data;
+        const signer = api.getSignerInstance(keypairVault.getKeyPair());
+        const activeAddress = getState().account.address;
+        /// dry run the transaction to get the effects
+        const [dryRunResponse, txGasEstimation] = await Promise.all([
+            signer.dryRunTransaction(txData),
```

Comment:
You can get the gas cost from the dry run result. No need to call getGasCostEstimation that does another dry run. (Just to save one call) just see how getGasCostEstimation calculates it and do it here

Reply:
did not check, just moved over the function, I will update